modules/broker-core/src/main/java/org/wso2/broker/core/ExchangeRegistry.java

```diff
@@ -85,34 +99,61 @@ void declareExchange(String exchangeName, Exchange.Type type,
                     "Exchange [ " + exchangeName
                             + " ] doesn't exists. Passive parameter is set, hence not creating the exchange.");
         } else if (exchange == null) {
-            exchange = ExchangeFactory.newInstance(exchangeName, type);
+            BindingDao dao = durable ? bindingDao : NO_OP_BINDING_DAO;
+            exchange = ExchangeFactory.newInstance(exchangeName, type, dao);
             exchangeMap.put(exchange.getName(), exchange);
-        } else if (!passive && exchange.getType() != type) { // TODO add durable check
+            if (durable) {
+                exchangeDao.persist(exchange);
+            }
+        } else if (!passive) {
             throw new BrokerException("Exchange [ " + exchangeName + " ] already exists.");
         } else if (exchange.getType() != type) {
             throw new BrokerException("Exchange type [ " + type + " ] does not match the existing one [ "
-                                              + exchange.getType() + " ].");
+                    + exchange.getType() + " ].");
+        }
+    }
+
+    private void retrieveAllExchangesFromDao() throws BrokerException {
+        exchangeDao.retrieveAll(
+                (name, typeString) -> {
+                    Exchange exchange = ExchangeFactory.newInstance(name,
+                            Exchange.Type.from(typeString), bindingDao);
+                    exchangeMap.putIfAbsent(name, exchange);
+                });
+    }
+
+    public Exchange getDefaultExchange() {
+        return defaultExchange;
```

Comment:
Can we rewrite this like below and remove the defaultExchange field?
```java
        return exchangeMap.get(DEFAULT);
```

Reply:
Updated.